stan/math/prim/fun/value_of.hpp

```diff
@@ -40,71 +41,11 @@ inline double value_of(const T x) {
  * @param x value
  * @return input value
  */
-template <>
-inline double value_of<double>(double x) {
-  return x;
+template <typename T, require_double_or_int_t<scalar_type_t<T>>...>
+inline auto value_of(T&& x) {
```

Comment:
```suggestion
inline decltype(auto) value_of(T&& x) {
```

link on usage of `decltype(auto)`

https://stackoverflow.com/a/24109800/2269255

Reply:
Oh whoops I got confused by the forward below. I'd either just use auto and make a copy of the int/double without forward or use decltype(auto) with forwarding.

Reply:
This one is also for any containers of int/double (or containers of containers), so I'll probably go with decltype(auto) and forwarding